src/libraries/System.Runtime.InteropServices.RuntimeInformation/src/System/Runtime/InteropServices/RuntimeInformation/RuntimeInformation.Unix.cs

```diff
@@ -26,69 +26,79 @@ public static Architecture OSArchitecture
         {
             get
             {
-                lock (s_osLock)
+                Debug.Assert(sizeof(Architecture) == sizeof(int));
+
+                if (s_osArch == -1)
```

Comment:
Technically I believe this still isn't thread-safe because s_osArch isn't volatile and, while it may not happen in practice, I believe it's valid on some architectures for a read to be introduced that would effectively make the code more akin to:
```C#
int arch = s_osArch;
if (s_osArch != -1) return arch;
...
```
in which case the first read could see -1 and -1 could be returned.